plugin/src/index.ts

```diff
@@ -0,0 +1,194 @@
+    addReplacingMainApplicationMetaDataItem(
+      config.modResults,
+      'com.google.android.gms.ads.flag.OPTIMIZE_AD_LOADING',
+      optimizeAdLoading.toString(),
+    );
+
+    return config;
+  });
+};
+
+const withIosAppId: ConfigPlugin<PluginParameters['iosAppId']> = (config, iosAppId) => {
+  if (iosAppId === undefined) return config;
+
+  return withInfoPlist(config, config => {
+    config.modResults.GADApplicationIdentifier = iosAppId;
+    return config;
+  });
+};
+
+const withIosAppMeasurementInitDelayed: ConfigPlugin<
+  PluginParameters['delayAppMeasurementInit']
+> = (config, delayAppMeasurementInit = false) => {
+  return withInfoPlist(config, config => {
+    config.modResults.GADDelayAppMeasurementInit = delayAppMeasurementInit;
+    return config;
+  });
+};
+
+const withIosSkAdNetworkItems: ConfigPlugin<PluginParameters['skAdNetworkItems']> = (
+  config,
+  skAdNetworkItems,
+) => {
+  if (skAdNetworkItems === undefined) return config;
+
+  return withInfoPlist(config, config => {
+    config.modResults.SKAdNetworkItems = config.modResults.SKAdNetworkItems ?? [];
+
+    const existingIdentifiers = config.modResults.SKAdNetworkItems.map(
+      (item: { SKAdNetworkIdentifier: string }) => item.SKAdNetworkIdentifier,
+    );
+
+    const missingIdentifiers = skAdNetworkItems.filter(
+      skAdNetworkItem => !existingIdentifiers.includes(skAdNetworkItem),
+    );
+
+    config.modResults.SKAdNetworkItems.push(
+      ...missingIdentifiers.map(identifier => ({
+        SKAdNetworkIdentifier: identifier,
+      })),
+    );
+
+    return config;
+  });
+};
+
+const withIosUserTrackingUsageDescription: ConfigPlugin<
+  PluginParameters['userTrackingUsageDescription']
+> = (config, userTrackingUsageDescription) => {
+  if (userTrackingUsageDescription === undefined) return config;
+
+  return withInfoPlist(config, config => {
+    config.modResults.NSUserTrackingUsageDescription = userTrackingUsageDescription;
+    return config;
+  });
+};
+
+const withReactNativeGoogleMobileAds: ConfigPlugin<PluginParameters> = (
+  config,
+  {
+    androidAppId,
+    delayAppMeasurementInit,
+    optimizeInitialization,
+    optimizeAdLoading,
+    iosAppId,
+    skAdNetworkItems,
+    userTrackingUsageDescription,
+  } = {},
+) => {
+  if (androidAppId === undefined) {
+    console.warn(
+      "No 'androidAppId' was provided. The native Google Mobile Ads SDK will crash on Android without it.",
+    );
+  }
+
+  if (iosAppId === undefined) {
+    console.warn(
+      "No 'iosAppId' was provided. The native Google Mobile Ads SDK will crash on iOS without it.",
+    );
+  }
+
+  return withPlugins(config, [
+    // Android
+    [withAndroidAppId, androidAppId],
+    [withAndroidAppMeasurementInitDelayed, delayAppMeasurementInit],
+    [withAndroidInitializationOptimized, optimizeInitialization],
+    [withAndroidAdLoadingOptimized, optimizeAdLoading],
+    // iOS
+    [withIosAppId, iosAppId],
```

Comment:
Is this the recommended way to have a function per entry?
Or could we also have one function that changes all info.plist entries?

Reply:
It's recommended to have separate entries/hooks for platforms and _features_. I felt like these are all kinda separate features so I went with this granular approach.

(I was also hoping to be able to test the hooks individually this way, but the Expo tooling for config plugin testing was kinda disappointing. I started working on this upstream in Expo).